lib/controllers/comment-decorations-controller.js

```diff
@@ -0,0 +1,231 @@
+
+    // only show comment decorations if we're on a checked out pull request
+    // otherwise, we'd have no way of knowing which comments to show.
+    if (
+      !this.isCheckedOutPullRequest(
+        this.props.repoData.branches,
+        this.props.repoData.remotes,
+        pullRequest,
+      )
+    ) {
+      return null;
+    }
+
+    const threadDataByPath = new Map();
+    const workdirPath = this.props.repoData.workingDirectoryPath;
+
+    for (const {comments, thread} of this.props.commentThreads) {
+      // Skip comment threads that are entirely minimized.
+      if (comments.every(comment => comment.isMinimized)) {
+        continue;
+      }
+
+      // There may be multiple comments in the thread, but we really only care about the root comment when rendering
+      // decorations.
+      const threadData = {
+        rootCommentID: comments[0].id,
+        threadID: thread.id,
+        position: comments[0].position,
+        nativeRelPath: toNativePathSep(comments[0].path),
+        fullPath: path.join(workdirPath, toNativePathSep(comments[0].path)),
+      };
+
+      if (threadDataByPath.get(threadData.fullPath)) {
+        threadDataByPath.get(threadData.fullPath).push(threadData);
+      } else {
+        threadDataByPath.set(threadData.fullPath, [threadData]);
+      }
+    }
+
+    /* istanbul ignore next */
+    if (threadDataByPath.size === 0) {
+      return null;
+    }
+
+    const openEditorsWithCommentThreads = this.getOpenEditorsWithCommentThreads(threadDataByPath);
+    return openEditorsWithCommentThreads.map(editor => {
+      const threadData = threadDataByPath.get(editor.getPath());
+      const translations = this.props.commentTranslations.get(threadData[0].nativeRelPath);
+
+      return (
+        <Fragment key={`github-editor-decoration-${editor.id}`}>
+          <Gutter
+            name="github-comment-icon"
+            priority={1}
+            className="comment"
+            editor={editor}
+            type="decorated"
+          />
+          <EditorCommentDecorationsController
+            endpoint={this.props.endpoint}
+            owner={this.props.owner}
+            repo={this.props.repo}
+            number={pullRequest.number}
+            workdir={workdirPath}
+            workspace={this.props.workspace}
+            editor={editor}
+            fileName={editor.getPath()}
+            headSha={pullRequest.headRefOid}
+            threadsForPath={threadData}
+            commentTranslationsForPath={translations}
+          />
+        </Fragment>
+      );
+    });
+  }
+
+  getOpenEditorsWithCommentThreads(threadDataByPath) {
+    const haveThreads = [];
+    for (const editor of this.state.openEditors) {
+      if (threadDataByPath.has(editor.getPath())) {
+        haveThreads.push(editor);
+      }
+    }
+    return haveThreads;
+  }
+
+  // Determine if we already have this PR checked out.
+  // todo: if this is similar enough to pr-checkout-controller, extract a single
```

Comment:
Not sure if addressing this TODO is in scope for this pr or not -- what do you think?

Reply:
Filed as #2045 :ok_hand: